### PR TITLE
Db shrink optimization

### PR DESCRIPTION
**Pull request by tbssajal**
tested

### CODE REVIEW

File: src/Lachain.Storage/DbCompact/DbShrinkRepository.cs [hunk 1]
```diff
@@ -130,6 +133,45 @@ public void DeleteVersion(uint repository, ulong block, ulong version)
             );
         }
 
+        private void SetTimePassed(ulong timePassed)
```
Comment (sgladkov) [inline src/Lachain.Storage/DbCompact/DbShrinkRepository.cs:136]: Should we save benchmark data in database?

**Review by sgladkov — COMMENTED** [review]
Does this db changes compatible with the old database? Have you tested update from the current dev chain to this new one?

File: src/Lachain.Core/CLI/CommandLineOptions.cs [hunk 2]
```diff
@@ -117,6 +117,6 @@ public class DbOptions
         public ulong depth { get; set; } = 100;
         
         [Option('h', "consistencyCheck", Required = false, HelpText = "Consistency check after dbCompact \"hard\"")]
-        public bool consistencyCheck { get; set; } = false;
+        public string consistencyCheck { get; set; } = "false";
```
[the remark this answers is not in the document]
Comment (tbssajal) [reply src/Lachain.Core/CLI/CommandLineOptions.cs:120]: giving no parameters take false but if i use --h false, it was not working for me. that is why. can you check if bool works?

File: src/Lachain.Storage/DbCompact/DbShrinkRepository.cs (hunk 1 above)
Comment (tbssajal) [reply src/Lachain.Storage/DbCompact/DbShrinkRepository.cs:136]: not sure about that. the process takes a lot of time. node can shut down in the middle, so I put it is db.

**Comment by tbssajal** [comment]
> Are this db changes compatible with the old database? Have you tested update from the current dev chain to this new one?

sorry, forgot about it. will test

File: src/Lachain.Core/CLI/CommandLineOptions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgladkov) [reply src/Lachain.Core/CLI/CommandLineOptions.cs:120]: https://stackoverflow.com/questions/35873835/command-line-parser-library-boolean-parameter

**Comment by tbssajal** [comment]
> > Are this db changes compatible with the old database? Have you tested update from the current dev chain to this new one?
> 
> sorry, forgot about it. will test

tested

File: src/Lachain.Storage/DbCompact/DbShrinkRepository.cs (hunk 1 above)
Comment (sgladkov) [reply src/Lachain.Storage/DbCompact/DbShrinkRepository.cs:136]: ok,  let's keep it in database

File: src/Lachain.Core/CLI/CommandLineOptions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbssajal) [reply src/Lachain.Core/CLI/CommandLineOptions.cs]: Thanks!

**Review by sgladkov — APPROVED** [review]

**Pull request merged**